go/libraries/doltcore/sqle/dolt_diff_summary_table_function.go

```diff
@@ -0,0 +1,503 @@
+func (ds *DiffSummaryTableFunction) WithChildren(children ...sql.Node) (sql.Node, error) {
+	if len(children) != 0 {
+		panic("unexpected children")
+	}
+	return ds, nil
+}
+
+// CheckPrivileges implements the interface sql.Node.
+func (ds *DiffSummaryTableFunction) CheckPrivileges(ctx *sql.Context, opChecker sql.PrivilegedOperationChecker) bool {
+	if ds.tableNameExpr != nil {
+		if !sql.IsText(ds.tableNameExpr.Type()) {
+			return false
+		}
+
+		tableNameVal, err := ds.tableNameExpr.Eval(ds.ctx, nil)
+		if err != nil {
+			return false
+		}
+		tableName, ok := tableNameVal.(string)
+		if !ok {
+			return false
+		}
+
+		// TODO: Add tests for privilege checking
+		return opChecker.UserHasPrivileges(ctx,
+			sql.NewPrivilegedOperation(ds.database.Name(), tableName, "", sql.PrivilegeType_Select))
+	}
+
+	tblNames, err := ds.database.GetTableNames(ctx)
+	if err != nil {
+		return false
+	}
+
+	var operations []sql.PrivilegedOperation
+	for _, tblName := range tblNames {
+		operations = append(operations, sql.NewPrivilegedOperation(ds.database.Name(), tblName, "", sql.PrivilegeType_Select))
+	}
+
+	return opChecker.UserHasPrivileges(ctx, operations...)
+}
+
+// Expressions implements the sql.Expressioner interface.
+func (ds *DiffSummaryTableFunction) Expressions() []sql.Expression {
+	exprs := []sql.Expression{ds.fromCommitExpr, ds.toCommitExpr}
+	if ds.tableNameExpr != nil {
+		exprs = append(exprs, ds.tableNameExpr)
+	}
+	return exprs
+}
+
+// WithExpressions implements the sql.Expressioner interface.
+func (ds *DiffSummaryTableFunction) WithExpressions(expression ...sql.Expression) (sql.Node, error) {
+	if len(expression) < 2 || len(expression) > 3 {
+		return nil, sql.ErrInvalidArgumentNumber.New(ds.FunctionName(), "2 or 3", len(expression))
+	}
+
+	for _, expr := range expression {
+		if !expr.Resolved() {
+			return nil, ErrInvalidNonLiteralArgument.New(ds.FunctionName(), expr.String())
+		}
+	}
+
+	ds.fromCommitExpr = expression[0]
+	ds.toCommitExpr = expression[1]
+	if len(expression) == 3 {
+		ds.tableNameExpr = expression[2]
+	}
+
+	// validate the expressions
+	if !sql.IsText(ds.fromCommitExpr.Type()) {
+		return nil, sql.ErrInvalidArgumentDetails.New(ds.FunctionName(), ds.fromCommitExpr.String())
+	}
+
+	if !sql.IsText(ds.toCommitExpr.Type()) {
+		return nil, sql.ErrInvalidArgumentDetails.New(ds.FunctionName(), ds.toCommitExpr.String())
+	}
+
+	if ds.tableNameExpr != nil {
+		if !sql.IsText(ds.tableNameExpr.Type()) {
+			return nil, sql.ErrInvalidArgumentDetails.New(ds.FunctionName(), ds.tableNameExpr.String())
+		}
+	}
+
+	return ds, nil
+}
+
+// RowIter implements the sql.Node interface
+func (ds *DiffSummaryTableFunction) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error) {
+	fromCommitVal, toCommitVal, tableName, err := ds.evaluateArguments()
+	if err != nil {
+		return nil, err
+	}
+
+	sqledb, ok := ds.database.(Database)
+	if !ok {
+		panic(fmt.Sprintf("unexpected database type: %T", ds.database))
```

Comment:
Turn this panic into an error, too. Even though it's something really unlikely (until we add a new type of database, which I'm sure we eventually will...) returning an error gets you the same effect you want (stopping execution and communicating the problem) without the danger of bringing down the multi-user sql-server.